spinn_machine/machine.py

```diff
@@ -645,3 +645,21 @@ def total_cores(self):
         """
         return len([
             processor for chip in self.chips for processor in chip.processors])
+
+    def has_wrap_arounds(self):
+        """ If the machine has wrap around links
+
+        :return: True if wrap around links exist, false otherwise
+        :rtype: bool
+        """
+        if self.max_chip_x == 2 and self.max_chip_y == 2:
+            return True
+        elif self.max_chip_x == 8 and self.max_chip_y == 8:
+            return False
+        elif self.max_chip_x % 12 == 0 and self.max_chip_y % 12 == 0:
+            return True
+        elif ((self.max_chip_x - 4) % 12 == 0
+                and (self.max_chip_y - 4) % 12 == 0):
+            return False
+        else:
+            return False
```

Comment:
This could all really be simplified to only check for the 2x2 case and the %12 x %12 case e.g.:
```
if 2x2 board or %12 x %12 board:
    return True
return False
```